.github/workflows/all.yml

```diff
@@ -226,19 +226,34 @@ jobs:
             docker run $DOCKER_IMAGE_ID scale --help
             docker run $DOCKER_IMAGE_ID pause --help
             docker run $DOCKER_IMAGE_ID resume --help
+
       - name: Publish
-        if: github.event_name != 'pull_request'
+        if: |
+          github.ref == 'refs/heads/master' ||
+          startsWith(github.ref, 'refs/tags/v')
         run: |
-          echo "REF=${{ github.ref }}"
-          echo "DOCKER_IMAGE_ID=$DOCKER_IMAGE_ID"
-          # Log into registry
+          echo "Publish GHCR ($DOCKER_IMAGE_ID:$VERSION)"
+
+          # Log into GitHub Container Registry
+          echo "${{ secrets.CR_PAT }}" | docker login https://ghcr.io -u ${{ github.actor }} --password-stdin
+          docker tag "$DOCKER_IMAGE_ID" "ghcr.io/$DOCKER_IMAGE_ID:$VERSION"
+          docker push "ghcr.io/$DOCKER_IMAGE_ID:$VERSION"
+          # We also want to tag the latest stable version as latest
+          if [[ "$VERSION" != "master" ]] && [[ ! "$VERSION" =~ (RC|rc) ]]; then
+            echo 'Publish GHCR (latest)'
+            docker tag "$DOCKER_IMAGE_ID" "ghcr.io/$DOCKER_IMAGE_ID:latest"
+            docker push "ghcr.io/$DOCKER_IMAGE_ID:latest"
+          fi
+
+          echo "Publish Docker ($DOCKER_IMAGE_ID:$VERSION)"
+
+          # Log into Docker Hub Registry
           echo "${{ secrets.DOCKER_PASS }}" | docker login -u "${{ secrets.DOCKER_USER }}" --password-stdin
-          VERSION="${VERSION#v}"
-          echo "VERSION=$VERSION"
```

Comment:
I think this is still needed for both GHCR and Docker Hub, as it strips the `v` prefix from the version tag, which is how we've been tagging images so far.